### ISSUE TITLE
Elastic Agent Plugin can no longer create new agents after pod restart (or 1 hour)

### ISSUE THREAD

**Issue by andcri** [gocd/helm-chart#98]
Hello, 
I m using the latest version of this plugin trough the gocd helm chart, when I deploy the helm chart for the first time everything works fine but when I restart the pod (or the pod gets restarted by a helm update) I get the following error

```
2024-01-05 10:36:54,589  INFO [167@MessageListener for ServerPingListener] p.c.g.c.e.k.c.g.c.e.KubernetesPlugin:72 [plugin-cd.go.contrib.elasticagent.kubernetes] - Did not find any running kubernetes pods.
2024-01-05 10:37:54,579 ERROR [167@MessageListener for ServerPingListener] p.c.g.c.e.k.c.g.c.e.KubernetesPlugin:127 [plugin-cd.go.contrib.elasticagent.kubernetes] - Error occurred while trying to list kubernetes pods:
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://172.20.0.1:443/api/v1/namespaces/gocd/pods. Message: Unauthorized. Received status: Status(apiVersion=v1, code=401, details=null, kind=Status, message=Unauthorized, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=Unauthorized, status=Failure, additionalProperties={}).

```

Is this the expected behavior when running in a cluster? do I need to add extra configuration in order to avoid this? Thanks

**Comment by chadlwilson** [comment]
The root of this problem is now fixed with chart [2.6.0](https://github.com/gocd/helm-chart/releases/tag/gocd-2.6.0) (which defaults to plugin [v4.0.0-505](https://github.com/gocd/kubernetes-elastic-agents/releases/tag/v4.0.0-505)) if you want to give it a go.

Note that the server configuration only happens once, so you'd either need to start with a fresh server (probably not ideal), or upgrade the plugin via the Helm chart, and then clear out the older configuration values from the Cluster Profile so it falls back to the defaults that the Helm chart ensures are set up.

**Comment by andcri** [comment]
Thank you very much for the quick fix and the explanation, works perfectly now 👍
